app/controllers/publish/users_edit_check_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Publish
+  class UsersEditCheckController < PublishController
+    before_action :authorize_provider
+
+    def show
+      @user_form = UserForm.new(current_user, user)
+    end
+
+    def update
+      @user_form = UserForm.new(current_user, user)
+      if @user_form.save!
+        UserAssociationsService::Create.call(user: @user_form.model, provider:) if @user_form.model.providers.exclude?(provider)
+        authorize(provider)
```

Comment:
Redundant? You're already calling it in the before action. Also feels like the `UserAssociationsService` should be tucked into the user form since it appears to be a dependancy of it?

Reply:
Removed redundnant authorisation here: https://github.com/DFE-Digital/publish-teacher-training/pull/3185/commits/1014c6a2180f4c91c11bce1e129da885e3b467bf

What do you mean "should be tucked into the user form"?

Reply:
Discussed offline - will optimise later.